src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,192 @@
+    private static final int AGE_UNDER_18 = 10;
+    private static final int NEGATIVE_AGE = -1;
+    private static final int MAX_AGE = 105;
+    private static final int OVER_MAX_AGE = 106;
+    private static final int AGE_OVER_18 = 20;
+    private static final String USER_NAME = "Docent";
+    private static final String NEW_USER_NAME = "Daisy";
+    private static final String USER_PASSWORD_OK = "123456";
+    private static final String USER_PASSWORD_THREE_CHARS = "123";
+    private static final String USER_PASSWORD_FIVE_CHARS = "12345";
+    private static final String USER_PASSWORD_LENGTH_OK = "1234567";
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final long USER_ID = 123L;
+    private static RegistrationService registrationService;
+    private static User user;
+    private static User newUser;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin(USER_NAME);
+        user.setAge(AGE_18);
+        user.setPassword(USER_PASSWORD_OK);
+
+        newUser = new User();
+        newUser.setLogin(USER_NAME);
+        newUser.setAge(AGE_18);
+        newUser.setPassword(USER_PASSWORD_OK);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        user = null;
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+            throw new RegistrationException(" ");
+        });
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        user.setLogin(" ");
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+            throw new RegistrationException("not allowed empty line!!!");
+        });
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+            throw new RegistrationException("empty line is not ok, please pass any login");
+        });
+    }
+
+    @Test
+    void register_sameLogin_notOk() {
+        newUser.setLogin(USER_NAME);
+        assertEquals(newUser.getLogin(), user.getLogin());
+    }
+
+    @Test
+    void register_rightLogin_ok() {
+        user.setLogin(NEW_USER_NAME);
+        assertNotEquals(newUser.getLogin(), user.getLogin());
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void register_negativeAge_notOk() {
+        user.setAge(NEGATIVE_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);;
+        });
+    }
+
+    @Test
+    void register_underAge_notOk() {
+        user.setAge(AGE_UNDER_18);

```

Comment:
**edge case** is not tested
![image](https://github.com/mate-academy/jv-registration-validation/assets/20326858/bd0b2c94-5579-4dd9-8735-2ea97c616bad)